lux/lib/lux/llm/together_ai.ex

```diff
@@ -0,0 +1,284 @@
+defmodule Lux.LLM.TogetherAI do
+    @moduledoc """
+    Together AI LLM implementation that supports passing Beams, Prisms, and Lenses as tools.
+
+    ## Configuration
+
+    Add to your config/runtime.exs:
+    ```elixir
+    config :lux,
+      api_keys: [
+        together: System.get_env("TOGETHER_API_KEY")
```

Comment:
We shouldn't use or reccomend System.get_env directly, especially at compile time. Instead, how updating and using [Lux.Config](https://github.com/Spectral-Finance/lux/blob/ba0c01b9d913bf7e836afdfc1adefc1f56efd399/lux/lib/lux/config.ex#L1)?